### PR TITLE
3339711 Add POTS to Sous

### PR DESCRIPTION
**Pull request by robert-arias**
## Issue:
- Add ~~Project CI~~ [POTS](https://github.com/fourkitchens/pots) to Sous
- DO Issue: https://www.drupal.org/project/sous/issues/3339711

## Description:
- Add ~~project ci~~ POTS to Sous and minimum requirements so a new build only needs to configure their ssh keys, tokens, etc on their deployment environments.

## Testing:
- Run `composer install` and make sure ~~project ci~~ POTS is installed.
- Follow [docs for pots](https://github.com/fourkitchens/pots#installation) to configure an environment on Pantheon and deploy a test project with Sous using POTS.

## Notes

- In order for Sous to fully function out of the box with Pantheon it would require committing a `/web` directory and `composer.lock` file. We decided this was not the best idea because we do not want to track these things in our repo.
- This PR was simplified to only include what POTS expects Sous to have in order to work with Circle Ci and Pantheon.
- We also discussed maybe the `pantheon.yml` gets scaffolded only if it is a Pantheon site. This PR doesn't include this functionality but does include the `pantheon.yml` file with settings that match Sous.
- While we don't have lint configured out of the box; I did provide the `lint` composer command and added a `lint.sh` script file to avoid linting errors since POTS will expect something here.
- I saw code sniffing wasn't working either so I installed the necessary packages and pointed the composer `code-sniff` command straight to phpcs and phpcbf without the added args and options. We can always improve on this but at least it runs now.
- I added a phpcs.xml.dist which is expected for code-sniff. You will also notice I excluded checking emulsify.php because it fails and I didn't think we should sort that out with this PR.

### CODE REVIEW

**Comment by JRuiz28** [comment]
Work well for me! 👍🏾 maybe @ccjjmartin can confirm if it is what is needed

File: composer.json [hunk 1]
```diff
@@ -19,12 +19,13 @@
...
-        "fourkitchens/sous-drupal-distro": "4.x-dev",
-        "drupal/config_direct_save": "^1.0",
         "cweagans/composer-patches": "^1.6.5",
-        "drupal/core-recommended": "^9",
+        "drupal/config_direct_save": "^1.0",
         "drupal/core-composer-scaffold": "^9",
+        "drupal/core-recommended": "^9",
         "drush/drush": "^10.0",
+        "fourkitchens/project_ci": "^0.4.0",
+        "fourkitchens/sous-drupal-distro": "4.x-dev",
```
Comment (ccjjmartin) [inline composer.json:28]: This should be removed now that the distro is gone.

**Comment by ccjjmartin** [comment]
Ideally, this PR would be associated with a site on Pantheon and the Project CI setup would include the necessary files to have CircleCI running the deployment.  This does assume we are going to support Pantheon out of the box though so let's discuss this in the next Sous meeting @robert-arias

**Comment by cbfannin** [comment]
@robert-arias @ccjjmartin This PR is marked "Needs Review" on d.o

Do you know where we left off with this?

**Comment by laura-johnson** [comment]
It would be good to get this set up to deploy to a demo site on Pantheon.

As part of this ticket, let's follow the steps here to [create a sous project](https://github.com/fourkitchens/project_ci) on Pantheon and configure this repo to deploy to it.

**Comment by cbfannin** [comment]
@laura-johnson @ccjjmartin I set up a [Pantheon site](https://dashboard.pantheon.io/sites/f23dede4-1941-4166-bf7e-7f5b65ce20b5#dev/code) but as I started the process to migrate our repo to Pantheon I got stuck with understanding what we need to do with the web directory. [Here is a thread on Slack of my confusion.](https://fourkitchens.slack.com/archives/CQJL99PB9/p1693317049655589)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,167 +1,57 @@
...
-
-
-### Features
-
-* add npm scripts to lando so ci and lando can run both seperately ([#102](https://github.com/fourkitchens/sous-drupal-project/issues/102)) ([412d607](https://github.com/fourkitchens/sous-drupal-project/commit/412d6073ea7b010c814748fc9cd6eadaa24db8d1))
-
-# [5.0.0](https://github.com/fourkitchens/sous-drupal-project/compare/4.4.0...5.0.0) (2023-04-21)
+# Changelog
 
+All notable changes specific to pantheon-upstreams/drupal-composer-managed are noted here.
```
Comment (laura-johnson) [inline CHANGELOG.md:3]: Why is the sous changelog overwritten with drupal-composer-managed changelog?

Comment (randallquesadaa) [reply CHANGELOG.md:3]: @laura-johnson Could it be a change from when @cbfannin merge the Pantheon repository with ours?

Comment (laura-johnson) [reply CHANGELOG.md:3]: Could be! I think we want the sous version though...

Comment (cbfannin) [reply CHANGELOG.md]: @laura-johnson @randallquesadaa Original changelog restored. Does that make this ready to merge? i.e. someone has functionally tested this?

**Comment by laura-johnson** [comment]
We have decided for now NOT to add POTS to Sous. Thanks for your great work on this everybody!

**Pull request closed without merging**